gomap_syn_scan.go

```diff
@@ -33,8 +37,40 @@ func sendSyn(laddr string, raddr string, sport uint16, dport uint16) error {
 		UrgentPointer: 0,
 	}
 
+	var conn net.Conn
+	var err error
+	var proxyDialer proxy.Dialer
+
 	// Connect to network interface to send packet
-	conn, err := net.Dial("ip4:tcp", raddr)
+	// use proxy dialer if proxyURL string is not empty
```

Comment:
If this code is referenced in duplicate multiple times then it should be moved to `gomap_funcs.go`

Reply:
ok done, refactored the code accordingly :) pls refer to this commit - https://github.com/JustinTimperio/gomap/pull/13/commits/28d830324dd74a66674eff513d36d5e8336646f1